includes/class-sensei-course.php

```diff
@@ -4195,6 +4197,45 @@ public function archive_page_content() {
 			remove_all_actions( 'sensei_pagination' );
 		}
 	}
+
+	/**
+	 * Take the user to login page if trying to access course results page without being logged in.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @access private
+	 *
+	 * @return void
+	 */
+	public function maybe_redirect_to_login_from_course_completion() {
+
+		if ( is_user_logged_in() ) {
+			return;
+		}
+
+		$settings = Sensei()->settings->get_settings();
+
+		$completed_page_id = intval( $settings['course_completed_page'] ?? 0 );
+
+		if ( $completed_page_id < 1 || get_the_ID() !== $completed_page_id ) {
```

Comment:
We should use Yoda conditions here as per https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#yoda-conditions

Reply:
Hi Donna 👋 I tried that at first, but got lint error from CI and had to change here https://github.com/Automattic/sensei/pull/6657/commits/fa365aae0e44d348a88bf5d78dd85fed1c5154b0. Also I get warning in the editor when I write that
![image](https://user-images.githubusercontent.com/6820724/226308822-fc48bd57-b3fb-4dcb-ad1e-c5fdfc324054.png)

Reply:
I was actually referring to the first condition in that `if` statement - `$completed_page_id < 1`. But it's OK, no biggie.